apps/prairielearn/src/lib/submissions-upload.sql

```diff
@@ -0,0 +1,91 @@
+-- BLOCK select_assessment_question
+SELECT
+  *
+FROM
+  assessment_questions
+WHERE
+  assessment_id = $assessment_id
+  AND question_id = $question_id;
+
+-- BLOCK insert_assessment_instance
+INSERT INTO
+  assessment_instances (assessment_id, user_id, number, open)
+VALUES
+  (
+    $assessment_id,
+    $user_id,
+    $instance_number,
+    false -- Assume closed by default when recreating
+  )
+RETURNING
+  id AS assessment_instance_id;
+
+-- BLOCK insert_instance_question
+INSERT INTO
+  instance_questions (
+    assessment_instance_id,
+    assessment_question_id,
+    requires_manual_grading,
+    status
+  )
+VALUES
+  (
+    $assessment_instance_id,
+    $assessment_question_id,
+    $requires_manual_grading,
+    'saved' -- Must not be the default 'unanswered' status
+  )
+RETURNING
+  id AS instance_question_id;
+
+-- BLOCK insert_variant
+INSERT INTO
+  variants (
+    instance_question_id,
+    question_id,
+    authn_user_id,
+    user_id,
+    variant_seed,
+    params,
+    true_answer,
+    options,
+    course_id
+  )
```

Comment:
Do you want to set the variant number? You have the info in the file.

Reply:
Done in 571fadc.